### PR TITLE
[Feat] Make user document async

### LINKED ISSUE #11150: ✨ Make user documents downloads async
## ♻️ Debt/Refactor

Move the application document to be generated asynchronously on the server.

## 🕵️ Details

We recently completed the generation of the user profile documents on the pool candidate tables but still need to do the same approach on the users table.

## 🙋‍♀️ Proposed Solution

Follow what was done in the [user profile document](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11071), creating a mutation and generating a document asynchronously but refactor where necessary to have it work on the users table.

## 🌎 Localization

All localization should exist in the front end, you just need to move missing ones to the server files.

## ✅ Acceptance Criteria

- [ ] Mutation created for user document download
- [ ] User document generator created
- [ ] Front end document generators removed and replaced with new mutation
- [ ] Event is fired, queued in job and notification is sent to the appropriate user

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #11150 

## 👋 Introduction

Makes the user profile documents async.

## 🕵️ Details

This shuffles some stuff around to reuse the code from the candidate user profile doc.

## 🧪 Testing

1. Build app `pnpm run dev:fresh`
2. Start worker queue `make queue-work`
3. Login as admin `admin@test.com`
4. Navigate to users table `/admin/users`
5. Select some users
6. Download the profile documents
7. Confirm job is queued and notification generated with link to proper file
8. Navigate to a users profile int he admin
9. Download their profile
10. Confirm job is queued and notification generated with link to proper file

### CODE REVIEW

**Comment by vd1992** [comment]
Sorry, but this might be a bit of a pain, but given that the user does not need a complete profile to be printed we will probably need a lot of nullsafe handling added.

Tested by creating a new account then attempting to print it, while admins are unlikely to print such accounts it is a handy way to test null safety for a lot of fields real quick

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
See above 😛

**Comment by esizer** [comment]
> Sorry, but this might be a bit of a pain, but given that the user does not need a complete profile to be printed we will probably need a lot of nullsafe handling added.

Goo catch! Should be better after [better handling for empty users](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/1a8595f16dfc4b11031a1af28b755500a7b7666f).

I created a new user and only did the minimum required fields and it seemed to work.

File: api/lang/fr/headings.php
Comment (vd1992) [inline api/lang/fr/headings.php:1]: Think this needs to be added to English headings too

image

File: api/app/Traits/Generator/GeneratesUserDoc.php [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+     */
+    protected function dei(Section $section, User $user, $headingRank = 4)
+    {
+        $section->addTitle($this->localizeHeading('dei'), $headingRank);
+
+        if ($user->indigenous_communities) {
+            foreach ($user->indigenous_communities as $community) {
+                $section->addListItem($this->localizeEnum($community, IndigenousCommunity::class, 'long'));
+            }
+        }
```
Comment (vd1992) [inline api/app/Traits/Generator/GeneratesUserDoc.php:221]: Something wrong here I think

image

File: api/app/Traits/Generator/GeneratesUserDoc.php [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+    /**
+     * Generate a users experiences
+     *
+     * @param  Section  $section  The section to add info to
+     * @param  User  $user  The user being generated
+     */
+    protected function experiences(Section $section, User $user, $headingRank = 3)
+    {
+
+        if ($user->experiences->count() > 0) {
```
Comment (vd1992) [inline api/app/Traits/Generator/GeneratesUserDoc.php:242]: Formatting of experiences is a little hard to parse, hard to tell where one begins and ends

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
A few more things, but looks almost there

File: api/lang/fr/headings.php
Comment (esizer) [reply api/lang/fr/headings.php:1]: Oops :sweat_smile: Added in [fix headings, experience spacing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/9ea766020d8987729f9b84b5050c074d5a5aa8db)

File: api/app/Traits/Generator/GeneratesUserDoc.php (hunk 1 above)
Comment (esizer) [reply api/app/Traits/Generator/GeneratesUserDoc.php]: Fixed in [fix headings, experience spacing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/9ea766020d8987729f9b84b5050c074d5a5aa8db)

File: api/app/Traits/Generator/GeneratesUserDoc.php (hunk 2 above)
Comment (esizer) [reply api/app/Traits/Generator/GeneratesUserDoc.php:242]: Oof, yeah that is bad. I tightened it up but it is still kinda hard to read :cry: I tihnk it is better at least :woman_shrugging: 

[fix headings, experience spacing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/9ea766020d8987729f9b84b5050c074d5a5aa8db)

File: api/app/Traits/Generator/GeneratesUserDoc.php (hunk 1 above)
Comment (vd1992) [reply api/app/Traits/Generator/GeneratesUserDoc.php]: The other ones work

image

But not this one, a problem? 🤔 

image

File: api/app/Traits/Generator/GeneratesUserDoc.php (hunk 2 above)
Comment (esizer) [reply api/app/Traits/Generator/GeneratesUserDoc.php:244]: lol you actually uncovered a bug :sweat_smile: Might look a bit better after [fix h5 style](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/de218ae9c47cb787a246e2611c0e959b04a8cee2)

File: api/app/Traits/Generator/GeneratesUserDoc.php (hunk 1 above)
Comment (esizer) [reply api/app/Traits/Generator/GeneratesUserDoc.php]: Looks like we actually are not rendering the legacy community on the current doc so it should be filtered out after [do not render legacy indigenous community](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/b12b6802a4cc6fd00457ad8ca7aca40ca93a31e1)

**Comment by vd1992** [comment]
Sorry, to squeeze in one more thing but I noticed this by chance. **Candidate** CSV download somehow got broken?

image

**Comment by esizer** [comment]
> Sorry, to squeeze in one more thing but I noticed this by chance. **Candidate** CSV download somehow got broken?

oh yeah, I moved the helper functions to a trait for re-use and forgot to add it to the CSV. Should be fixed after [add required trait to pool candidate CSV](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11151/commits/24a6980496f56e034f32fba488fb32a181068135)

File: api/app/Traits/Generator/GeneratesUserDoc.php (hunk 2 above)
Comment (vd1992) [reply api/app/Traits/Generator/GeneratesUserDoc.php:244]: Well that looks much nicer

**Review by vd1992 — APPROVED** [review]
Okay, I think it is good now. 
Great stuff!
